ext/opentelemetry-ext-docker-tests/tests/check_availability.py

```diff
@@ -0,0 +1,79 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import time
+import traceback
+
+import mysql.connector
+import pymongo
+
+MONGODB_COLLECTION_NAME = "test"
+MONGODB_DB_NAME = os.getenv("MONGODB_DB_NAME", "opentelemetry-tests")
+MONGODB_HOST = os.getenv("MONGODB_HOST", "localhost")
+MONGODB_PORT = int(os.getenv("MONGODB_PORT", "27017"))
+MYSQL_DB_NAME = os.getenv("MYSQL_DB_NAME ", "opentelemetry-tests")
+MYSQL_HOST = os.getenv("MYSQL_HOST ", "localhost")
+MYSQL_PORT = int(os.getenv("MYSQL_PORT ", "3306"))
+MYSQL_USER = os.getenv("MYSQL_USER ", "testuser")
+MYSQL_PASSWORD = os.getenv("MYSQL_PASSWORD ", "testpassword")
+RETRY_COUNT = 5
+RETRY_INTERVAL = 5  # Seconds
+
+
+def check_pymongo_connection():
+    # Try to connect to DB
+    for i in range(RETRY_COUNT):
+        try:
+            client = pymongo.MongoClient(
+                MONGODB_HOST, MONGODB_PORT, serverSelectionTimeoutMS=2000
+            )
+            db = client[MONGODB_DB_NAME]
+            collection = db[MONGODB_COLLECTION_NAME]
+            collection.find_one()
+            client.close()
+            break
+        except Exception as ex:
+            if i == RETRY_COUNT - 1:
+                raise (ex)
+            traceback.print_exc()
+        time.sleep(RETRY_INTERVAL)
+
+
+def check_mysql_connection():
+    # Try to connect to DB
+    for i in range(RETRY_COUNT):
+        try:
+            connection = mysql.connector.connect(
+                user=MYSQL_USER,
+                password=MYSQL_PASSWORD,
+                host=MYSQL_HOST,
+                port=MYSQL_PORT,
+                database=MYSQL_DB_NAME,
+            )
+            connection.close()
+            break
+        except Exception as ex:
+            if i == RETRY_COUNT - 1:
+                raise (ex)
```

Comment:
I see that this is being run outside of the standard testing process, so, I'll just add a suggestion here. I'm marking this approved to remove my blocking request for changes.

I don't find this to be very accurate. What we are doing here is printing every exception traceback but the last one and then raising the last one. Nevertheless, the last exception should not be treated in a different way than the rest because it decreased the attempt counter by one just like the rest.

If what we want is to inform the user of every exception that was raised, we should use [`exception`](https://docs.python.org/3/library/logging.html#logging.Logger.exception) instead of just printing it to console with `traceback.print_exc`. In that way, the logging of these exceptions can be handled in the standard way, like the rest of the logs.

Once the attempts are exhausted, we should raise an exception that informs the user precisely about this, something like:

```python
raise Exception("Unable to connect to database at {} after {} attempts".format(...))
```

In summary, something like this:

```python
from logging import getLogger

logger = getLogger(__name__)

attempts = 3

for attempt in range(3):
    try:
        1 / 0
        print("connected")
        break
    except Exception as error:
        logger.exception(error)
else:
    raise Exception(
        "Unable to print \"connected\" after {} attempts".format(attempts)
    )
```

```
division by zero
Traceback (most recent call last):
  File "example.py", line 9, in <module>
    1 / 0
ZeroDivisionError: division by zero
division by zero
Traceback (most recent call last):
  File "example.py", line 9, in <module>
    1 / 0
ZeroDivisionError: division by zero
division by zero
Traceback (most recent call last):
  File "example.py", line 9, in <module>
    1 / 0
ZeroDivisionError: division by zero
Traceback (most recent call last):
  File "example.py", line 16, in <module>
    "Unable to print \"connected\" after {} attempts".format(attempts)
Exception: Unable to print "connected" after 3 attempts
```

Reply:
Thanks for reviewing @ocelotl  I updated to use the logger instead of just printing the error, @toumorokoshi had some strong opinion about throwing our own exception instead of throwing last one, I will continue with this change to move this PR forward, this could be addressed in a future PR.